tsl/src/nodes/gapfill/planner.c

```diff
@@ -421,6 +426,16 @@ plan_add_gapfill(PlannerInfo *root, RelOptInfo *group_rel)
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 				 errmsg("multiple time_bucket_gapfill calls not allowed")));
 
+	/* Executing time_bucket_gapfill on distributed hypertable produces incorrect result,
+	 * returns internal planning error or crashes if grouping includes a space dimension.
+	 * Thus time_bucket_gapfill is temporary disabled until it is fixed.
+	 */
+	if (dist_ht)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("time_bucket_gapfill not supported distributed hypertable"),
+				 errdetail("This limitation might be removed in the future.")));
```

Comment:
I don't think we should make these kinds of statements in the code. It belongs to the documentation.

Reply:
It is suggestion from @erimatnor to replace my original statement, which was staying that the support is not implemented. I think it is important to state in the error message that it is not just not supported. It can insult users.

Reply:
See https://github.com/timescale/timescaledb/pull/2309#discussion_r482864128

Reply:
If find that a very strange argument.

Reply:
@mkindahl Can you elaborate?

Not supported means we are good to not support it. We often return this error for functionality, which we will never support or will not support in reasonable future. This case is that we postpone the support, so implicitly sending users to docs unlikely to help them with their frustration when they get the error.

Reply:
I find it very strange that we should write that we might support it in the future because if we do not say that we might insult somebody. That is at least how I read the argument.

Reply:
My suggestion was optional so I am fine to remove the errdetail if that resolves this.

Reply:
Since you discussed it already, please push. I have already approved the change.

Reply:
I changed the formulation, so it doesn't state explicitly that the limitation will removed. The new formulation still allows user to understand that it is a limitation.